src/main/scala/org/allenai/pdffigures2/FindGraphicsRaster.scala

```diff
@@ -57,7 +59,8 @@ object FindGraphicsRaster {
     pixThreshold: Int,
     rescale: Int
   ): List[Box] = {
-    val pixelsToExplore = scala.collection.mutable.Stack[Int]()
+    val pixelsToExplore = mutable.Set[Int]()
```

Comment:
The main insight is that the stack would keep the same pixel multiple times, often hundreds or thousands of times, because the way adjacent pixels are adjacent.